monkey/monkey_island/cc/agent_event_handlers/scan_event_handler.py

```diff
@@ -110,7 +109,7 @@ def _update_nodes(self, target_machine: Machine, event: ScanEvent):
 
     def _update_tcp_connections(self, src_node: Node, target_machine: Machine, event: TCPScanEvent):
         tcp_connections = set()
-        open_ports = (port for port, status in event.ports.items() if status == PortStatus.OPEN)
+        open_ports = self._get_open_ports(event)
```

Comment:
We can store the list of ports on line 52 and pass that list to this function instead of calling `_get_open_ports()` again.

Reply:
Information about open ports is already in the event. The way to go is to probably build a handler with unpacked values like target, open ports, etc. methods shouldn't have 10 parameters that are duplicate data.
Anyways, Mike said he'll have to refactor this anyway, so I'm not worried about multiple calls to the same method